### PR TITLE
[mds-policy] [mds-policy-author] Add global error handling middleware, improve error handling generally

### PR DESCRIPTION
**Pull request by avatarneil**
## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [x] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [x] Policy Author

### CODE REVIEW

File: packages/mds-policy-author/api.ts [hunk 1]
```diff
@@ -242,7 +242,7 @@ function api(app: express.Express): express.Express {
   /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
   app.use(async (error: Error, req: ApiRequest, res: ApiResponse) => {
     await logger.error(req.method, req.originalUrl, error)
-    return res.status(500).send(error)
+    return res.status(500).send({ error })
```
Comment (janedotx) [inline packages/mds-policy-author/api.ts:245]: You didn't want to do the fancy stuff with `next`?

Comment (avatarneil) [reply packages/mds-policy-author/api.ts:245]: That's precisely what's happening here, this _is_ the `next` function

File: packages/mds-policy/api.ts [hunk 2]
```diff
@@ -64,84 +65,105 @@ function api(app: express.Express): express.Express {
...
-      : { get_published: true }
-
-    if (start_date > end_date) {
-      res.status(400).send({ error: 'start_date after end_date' })
-      return
-    }
+  app.get(
+    pathsFor('/policies'),
+    async (req: PolicyApiRequest, res: GetPoliciesResponse, next: express.NextFunction) => {
+      // TODO extract start/end applicability
```
Comment (janedotx) [inline packages/mds-policy/api.ts:71]: Can these TODOs be deleted? It looks like they've been done.

File: packages/mds-policy/api.ts [hunk 3]
```diff
@@ -64,84 +65,105 @@ function api(app: express.Express): express.Express {
...
+        next(new ServerError(error))
+      }
     }
-  })
+  )
 
-  app.get(pathsFor('/policies/:policy_id'), async (req: PolicyApiRequest, res: GetPolicyResponse) => {
-    const { policy_id } = req.params
-    const { scopes } = res.locals
+  app.get(
```
Comment (janedotx) [inline packages/mds-policy/api.ts:120]: Still needs to be able to support pagination, according to the spec we want to submit.

Comment (avatarneil) [reply packages/mds-policy/api.ts:120]: Implementing pagination (properly) is not a small ask, I don't think we can get it done today.

File: packages/mds-policy/api.ts (hunk 2 above)
Comment (avatarneil) [reply packages/mds-policy/api.ts]: Yeup

**Pull request merged**